sinc backend implementation

pj-mathematician opened the pull request:
I am facing some issues with type casting, I added some explicit wrappers to ensure that the returned dtype is the same as that of the input dtype


This PR is a reference to the previous PR #4436 .

Issue is #3856

pj-mathematician commented:
Hey, can I get a review? I have added all the relevant functions in the backend and it seems that they are passing the tests too. The failed checks are not in the scope of my code, thanks!

johanjino commented:
@pj-mathematician Thanks for the contribution, and the work you've done seems amazing. Since there's various backends and tests, I'll keep checking them through.

In the meanwhile, it would great if you could add the docstrings for the functions as well since making it a different ToDo task increases the workflow. Moreover, since you have implemented them, you may be in good position to do this as well 😀.

pj-mathematician commented:
Thanks!! I will add them right away!

pj-mathematician commented:
> @pj-mathematician Thanks for the contribution, and the work you've done seems amazing. Since there's various backends and tests, I'll keep checking them through.
> 
> In the meanwhile, it would great if you could add the docstrings for the functions as well since making it a different ToDo task increases the workflow. Moreover, since you have implemented them, you may be in good position to do this as well 😀.

Added for ivy/array , ivy/container and ivy functional , with appropriate docs. Since sinc wasnt available on Python array API standard , I have added the mathematical constraints and special cases myself using the similar format. Let me know if there are any changes regarding this too! 🙌

pj-mathematician commented:
#4815 mentioning the issue to keep track

johanjino commented:
Hey @pj-mathematician , Im sorry this slipped my radar initially. Everything you have done is perfect. But since we are looking forward to a next stable release, we would like all the extension functions written in the `extensions.py` files so that they and built and then segregated until they are stable.
I would be great if you could just change the file under which the codes exist. I have made `extenstions.py` files wherever necessary🙂. I apologize for all the hassle.

pj-mathematician commented:
Hey! Thanks alot for your update, I will update accordingly and commit asap 😁

pj-mathematician commented:
@johanjino This is what i have understood after going through the `extensions.py` files:
I have to update the `functional/backends/*/extensions.py` and `functional/ivy/extensions.py`  by adding the same code with relevant imports. Do I have to change the tests in any way?

pj-mathematician commented:
8d4d25c :  
>  we would like all the extension functions written in the extensions.py files so that they and built and then segregated until they are stable.

Done 🙌🏻 and the elementwise files have been reverted back. Tested extensions.py locally and its passing for sinc 🙂

johanjino commented:
Hey @pj-mathematician , thanks for your contribution. This work is great!.
